src/main/java/jenkins/plugins/logstash/configuration/LogstashIndexer.java

```diff
@@ -48,6 +48,16 @@ public synchronized T getInstance()
     return instance;
   }
 
+  /**
+   * Purpose of this method is to validate the inputs (if required) and if found
+   * erroneous throw an exception so that it will be bubbled up to the UI.
+   * 
+   * @throws Exception
+   */
+  public void validate() throws Exception {
```

Comment:
[minor] it would be better to have something more specific than Exception in the signature (e.g. ConstraintViolationException); OTOH it would force us to catch and wrap MimeTypeParseException, so I'm not sure if it's worth it

Reply:
> OTOH it would force us to catch and wrap MimeTypeParseException, so I'm not sure if it's worth it

Thats the reason I didn't use specific exception there. In this case I feel we should not bother about the specific exception. If validate() throws any exception we know it is due to invalid indexer config so simply displaying wrapped exception would be sufficient . Let me know if you feel there is a value in that I can make a change.